src/v3/src/components/Form/renderUISchemaLayout.tsx

```diff
@@ -42,29 +56,43 @@ export const renderUISchemaLayout: any = (uischema: UISchemaLayout) => {
     >
       {
         elements.map((element, index) => {
-          if (isUISchemaLayoutType(element.type)) {
-            return renderUISchemaLayout(element as UISchemaLayout);
+          const elementKey = `${element.type}_${index}`;
+
+          if (element.type === UISchemaLayoutType.STEPPER) {
+            return (
+              <Stepper
+                key={elementKey}
+                uischema={element as StepperLayout}
+              />
+            );
           }
 
-          const renderer = renderers.find((r) => r.tester(element));
+          if ([UISchemaLayoutType.HORIZONTAL, UISchemaLayoutType.VERTICAL]
+            .includes((element as UISchemaLayout).type)) {
+            return (
+              <Layout
+                key={elementKey}
+                uischema={element}
+              />
+            );
+          }
 
+          const renderer = renderers.find((r) => r.tester(element));
           if (!renderer) {
             // TODO: for now do not render for unmatch render object
             // remove unnecessary uischema in future refactor and throw error
             // throw new Error(`Failed to find render component for uischema: ${JSON.stringify(element)}`);
-            if (!isProductionEnvironment()) {
+            if (isDevelopmentEnvironment() || isTestEnvironment()) {
               console.warn(`Failed to find render component for uischema: ${JSON.stringify(element)}`);
```

Comment:
I still not very confident if we will want to use v2 code in places that more than needed, eventually we most probably still want to get rid of those code to avoid Courage deps